### PR TITLE
SOC-1267 prepare tracking of each place where the force login modal is triggered

### PR DESCRIPTION
**Pull request by rafalkalinski**
https://wikia-inc.atlassian.net/browse/SOC-1267

### CODE REVIEW

File: extensions/wikia/AuthModal/js/AuthModal.js [hunk 1]
```diff
@@ -107,6 +115,10 @@ define('AuthModal', ['jquery', 'wikia.window'], function ($, window) {
 
 			if (window.wgEnableNewAuthModal) {
 				open(params.onAuthSuccess);
+				track({
+					action: Wikia.Tracker.ACTIONS.CLICK,
+					label: params.origin
```
Comment (kvas-damian) [inline extensions/wikia/AuthModal/js/AuthModal.js:123]: There is no `params` validation. Please make sure this attribute is always passed or use some default value here.

File: extensions/wikia/AuthModal/js/AuthModal.js [hunk 2]
```diff
@@ -67,6 +73,8 @@ define('AuthModal', ['jquery', 'wikia.window'], function ($, window) {
 			$blackout.remove();
 			isOpen = false;
 		}
+
+		$(window.document).add(window).off('.authModal');
```
Comment (bkoval) [inline extensions/wikia/AuthModal/js/AuthModal.js]: You're using document as window property and on line 28 as a global. Let's inject 'wikia.document' into this AMD module and then reference it in a consistent way :)

File: extensions/wikia/AuthModal/js/AuthModal.js [hunk 3]
```diff
@@ -25,9 +25,15 @@ define('AuthModal', ['jquery', 'wikia.window'], function ($, window) {
...
 		});
 
-		$(window.document).keyup(onKeyUp);
-		window.addEventListener('message', function (event) {
-			if (event.data.isUserAuthorized) {
+		$(window).on('keyup.authModal', document, onKeyUp);
+		$(window).on('message.authModal', function (event) {
+			var e = event.originalEvent;
+
+			if (typeof e.data === 'undefined' || typeof e.data.isUserAuthorized === 'undefined') {
```
Comment (bkoval) [inline extensions/wikia/AuthModal/js/AuthModal.js]: Is the second check needed? On line 36 you'll verify isUserAuthorized and everything else in this handler is inside the second "if"'s body.

File: extensions/wikia/AuthModal/js/AuthModal.js (hunk 2 above)
Comment (rafalkalinski) [inline extensions/wikia/AuthModal/js/AuthModal.js]: After recent changes we don't use document there anymore

**Comment by kvas-damian** [comment]
LGTM

**Comment by bkoval** [comment]
LGTM :palm_tree:

**Pull request merged**